shell/platform/windows/external_texture_gl.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/external_texture_gl.h"
+
+// OpenGL ES and EGL includes
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <EGL/eglplatform.h>
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+
+#include <iostream>
+
+namespace flutter {
+
+typedef void (*glGenTexturesProc)(GLsizei n, GLuint* textures);
+typedef void (*glDeleteTexturesProc)(GLsizei n, const GLuint* textures);
+typedef void (*glBindTextureProc)(GLenum target, GLuint texture);
+typedef void (*glTexParameteriProc)(GLenum target, GLenum pname, GLint param);
+typedef void (*glTexImage2DProc)(GLenum target,
+                                 GLint level,
+                                 GLint internalformat,
+                                 GLsizei width,
+                                 GLsizei height,
+                                 GLint border,
+                                 GLenum format,
+                                 GLenum type,
+                                 const void* data);
+
+struct GlProcs {
+  explicit GlProcs() {
+    glGenTextures =
+        reinterpret_cast<glGenTexturesProc>(eglGetProcAddress("glGenTextures"));
+    glDeleteTextures = reinterpret_cast<glDeleteTexturesProc>(
+        eglGetProcAddress("glDeleteTextures"));
+    glBindTexture =
+        reinterpret_cast<glBindTextureProc>(eglGetProcAddress("glBindTexture"));
+    glTexParameteri = reinterpret_cast<glTexParameteriProc>(
+        eglGetProcAddress("glTexParameteri"));
+    glTexImage2D =
+        reinterpret_cast<glTexImage2DProc>(eglGetProcAddress("glTexImage2D"));
+
+    valid = glGenTextures && glDeleteTextures && glBindTexture &&
+            glTexParameteri && glTexImage2D;
+  }
+
+  glGenTexturesProc glGenTextures;
+  glDeleteTexturesProc glDeleteTextures;
+  glBindTextureProc glBindTexture;
+  glTexParameteriProc glTexParameteri;
+  glTexImage2DProc glTexImage2D;
+  bool valid;
+};
+
+struct ExternalTextureGLState {
+  GLuint gl_texture = 0;
+  static const GlProcs procs;
```

Comment:
I think I'm missing something here. What is the point of a static const reference in the struct declaration when this is all private to this file anyway? Why not just reference the functions directly?